Sources/TensorFlow/Core/Utilities.swift

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
+import Darwin
+#else
+import Glibc
+#endif
+
+import CTensorFlow
+
+//===------------------------------------------------------------------------------------------===//
+// Runtime Checkers
+//===------------------------------------------------------------------------------------------===//
+
+/// These checks run in both debug and release modes (while assert() only runs in debug mode), to 
+/// help shake out more bugs and facilitate debugging in the early project phases. It can be
+/// replaced with plain assert() later, when we have a more mature code base.
+@inlinable
+public func internalConsistencyCheck(
+    _ predicate: Bool,
+    _ errMessage: String = "TF runtime assertion failure",
+    file: StaticString = #file,
+    line: UInt = #line
+) {
+    guard predicate else {
+        fatalError(errMessage, file: file, line: line)
+    }
+}
+
+@inlinable
+public func checkOk(
+    _ s: CTFStatus?,
+    file: StaticString = #file,
+    line: UInt = #line
+) {
+    internalConsistencyCheck(
+        TF_GetCode(s) == TF_OK,
+        String(cString: TF_Message(s)),
+        file: file,
+        line: line)
+}
+
+//===------------------------------------------------------------------------------------------===//
+// Type Aliases
+//===------------------------------------------------------------------------------------------===//
+
+// Before assigning a C pointer to one of the pointer type aliases below, caller should check that 
+// the pointer is not NULL.
+
+/// The `TF_Session *` type.
+public typealias CTFSession = OpaquePointer
```

Comment:
All utility and runtime types and functions should be internal.

Reply:
Done.